src/modules/navigator/MissionFeasibility/FeasibilityChecker.cpp

```diff
@@ -0,0 +1,692 @@
+	    mission_item.nav_cmd != NAV_CMD_SET_CAMERA_ZOOM &&
+	    mission_item.nav_cmd != NAV_CMD_SET_CAMERA_FOCUS &&
+	    mission_item.nav_cmd != NAV_CMD_DO_VTOL_TRANSITION) {
+
+		mavlink_log_critical(_mavlink_log_pub, "Mission rejected: item %i: unsupported cmd: %d\t",
+				     (int)(current_index + 1),
+				     (int)mission_item.nav_cmd);
+		events::send<uint16_t, uint16_t>(events::ID("navigator_mis_unsup_cmd"), {events::Log::Error, events::LogInternal::Warning},
+						 "Mission rejected: item {1}: unsupported command: {2}", current_index + 1, mission_item.nav_cmd);
+		return false;
+	}
+
+	/* Check non navigation item */
+	if (mission_item.nav_cmd == NAV_CMD_DO_SET_SERVO) {
+
+		/* check actuator number */
+		if (mission_item.params[0] < 0 || mission_item.params[0] > 5) {
+			mavlink_log_critical(_mavlink_log_pub, "Actuator number %d is out of bounds 0..5\t",
+					     (int)mission_item.params[0]);
+			events::send<uint32_t>(events::ID("navigator_mis_act_index"), {events::Log::Error, events::LogInternal::Warning},
+					       "Actuator number {1} is out of bounds 0..5", (int)mission_item.params[0]);
+			return false;
+		}
+
+		/* check actuator value */
+		if (mission_item.params[1] < -PWM_DEFAULT_MAX || mission_item.params[1] > PWM_DEFAULT_MAX) {
+			mavlink_log_critical(_mavlink_log_pub,
+					     "Actuator value %d is out of bounds -PWM_DEFAULT_MAX..PWM_DEFAULT_MAX\t", (int)mission_item.params[1]);
+			events::send<uint32_t, uint32_t>(events::ID("navigator_mis_act_range"), {events::Log::Error, events::LogInternal::Warning},
+							 "Actuator value {1} is out of bounds -{2}..{2}", (int)mission_item.params[1], PWM_DEFAULT_MAX);
+			return false;
+		}
+	}
+
+	// check if the mission starts with a land command while the vehicle is landed
+	if ((current_index == 0) && mission_item.nav_cmd == NAV_CMD_LAND && _is_landed) {
+
+		mavlink_log_critical(_mavlink_log_pub, "Mission rejected: starts with landing\t");
+		events::send(events::ID("navigator_mis_starts_w_landing"), {events::Log::Error, events::LogInternal::Info},
+			     "Mission rejected: starts with landing");
+		return false;
+	}
+
+	return true;
+}
+
+bool FeasibilityChecker::checkTakeoff(mission_item_s &mission_item)
+{
+
+	// look for a takeoff waypoint
+	if (mission_item.nav_cmd == NAV_CMD_TAKEOFF || mission_item.nav_cmd == NAV_CMD_VTOL_TAKEOFF) {
+		// make sure that the altitude of the waypoint is at least one meter larger than the acceptance radius
+		// this makes sure that the takeoff waypoint is not reached before we are at least one meter in the air
+
+		float takeoff_alt = mission_item.altitude_is_relative
+				    ? mission_item.altitude
+				    : mission_item.altitude - _home_alt_msl;
+
+		float acceptance_radius = _param_nav_acc_rad.get();
+
+		if (mission_item.acceptance_radius > NAV_EPSILON_POSITION) {
+			acceptance_radius = mission_item.acceptance_radius;
+		}
+
+		if (takeoff_alt - 1.0f < acceptance_radius) {
```

Comment:
This is such a random check, maybe time to remove or change it?

Reply:
@sfuhrer So just to be clear, we remove this check und thus also allow the user to upload the mission where the takeoff altitude is below the home altitude?

Reply:
I would simply check for takeoff_alt<FLT_EPS.